projects/netgrif-components-core/src/lib/data-fields/case-ref-field/abstract-case-ref-field.component.ts

```diff
@@ -0,0 +1,250 @@
+                }
+                default: {
+                    return undefined;
+                }
+            }
+        } else if (arc.type === ArcType.ARC) {
+            return this._petriflowFactoryService.createArc(RegularTransitionPlaceArc, source.canvasElement, destination.canvasElement, arc.breakpoints, arc.multiplicity);
+        } else {
+            return undefined;
+        }
+    }
+
+    protected createPetriflowArc(arc: ArcImport, newArc: Arc, source: PetriflowTransition | PetriflowPlace) {
+        if (source instanceof PetriflowPlace) {
+            switch (arc.type) {
+                case ArcType.ARC: {
+                    return this._petriflowFactoryService.createArc(PetriflowPlaceTransitionArc, newArc);
+                }
+                case ArcType.RESET: {
+                    return this._petriflowFactoryService.createArc(PetriflowResetArc, newArc);
+                }
+                case ArcType.INHIBITOR: {
+                    return this._petriflowFactoryService.createArc(PetriflowInhibitorArc, newArc);
+                }
+                case ArcType.READ: {
+                    return this._petriflowFactoryService.createArc(PetriflowReadArc, newArc);
+                }
+                default: {
+                    return undefined;
+                }
+            }
+        } else if (arc.type === ArcType.ARC) {
+            return this._petriflowFactoryService.createArc(PetriflowTransitionPlaceArc, newArc);
+        } else {
+            return undefined;
+        }
+    }
+
+    protected isEnabled(t: PetriflowTransition, places: Array<PetriflowPlace>, arcs: Array<PetriflowArc<any>>): boolean {
+        const testMarking: Map<string, number> = new Map();
+
+        for (const place of places) {
+            testMarking.set(place.canvasElement.id, place.canvasElement.tokensCount);
+        }
+        for (const arc of arcs) {
+            if (arc.element.end.id === t.canvasElement.id && (arc instanceof PetriflowInhibitorArc) && testMarking.get((arc.element.start as Place).id) >= parseInt(arc.element.multiplicity.data, 10)) {
+                return false;
+            }
+        }
+        for (const arc of arcs) {
+            if (arc.element.end.id === t.canvasElement.id && (arc instanceof PetriflowReadArc) && testMarking.get((arc.element.start as Place).id) < parseInt(arc.element.multiplicity.data, 10)) {
+                return false;
+            }
+        }
+        for (const arc of arcs) {
+            if (arc.element.end.id === t.canvasElement.id && arc instanceof PetriflowPlaceTransitionArc) {
+                const place = testMarking.get((arc.element.start as Place).id)
+                testMarking.set((arc.element.start as Place).id, place - parseInt(arc.element.multiplicity.data, 10));
+            }
+        }
+        for (const place of testMarking.values()) {
+            if (place < 0) {
```

Comment:
why does this check exist? negative marking should never exist

Reply:
explained in call, please close this question